src/env_object.cpp

```diff
@@ -19,24 +24,89 @@ Value EnvObject::inspect(Env *env) {
         hash->put(env, name, new StringObject { getenv(name->c_str()) });
         pair = *(environ + i);
     }
-    return hash->inspect(env);
+    return hash;
+}
+
+Value EnvObject::size(Env *env) const {
+    size_t idx = 0;
+    char *pair = *environ;
+    if (!pair) return Value::integer(0); // paranoia, unsure if necessary
+    for (; pair; idx++) {
+        pair = *(environ + idx);
+    }
+    return IntegerObject::from_size_t(env, idx);
+}
+
+Value EnvObject::delete_key(Env *env, Value name) {
+    name->assert_type(env, Object::Type::String, "String");
+    auto namestr = name->as_string()->c_str();
+    char *value = getenv(namestr);
+    if (value) {
+        ::unsetenv(namestr);
+        return new StringObject { value };
+    } else {
+        return NilObject::the();
+    }
 }
 
+Value EnvObject::assoc(Env *env, Value name) {
+    StringObject *namestr;
+    namestr = name->is_string() ? name->as_string() : name->to_str(env);
+    char *value = getenv(namestr->c_str());
+    if (value) {
+        StringObject *valuestr = new StringObject { value };
+        return new ArrayObject { { namestr, valuestr } };
+    } else {
+        return NilObject::the();
+    }
+}
 Value EnvObject::ref(Env *env, Value name) {
+    StringObject *namestr;
+    namestr = name->is_string() ? name->as_string() : name->to_str(env);
+    char *value = getenv(namestr->c_str());
+    if (value) {
+        return new StringObject { value };
+    } else {
+        return NilObject::the();
+    }
+}
+
+Value EnvObject::fetch(Env *env, Value name) {
     name->assert_type(env, Object::Type::String, "String");
     char *value = getenv(name->as_string()->c_str());
     if (value) {
         return new StringObject { value };
     } else {
-        return NilObject::the();
+        env->raise("KeyError", "key not found: {}", name->as_string()->string());
     }
 }
 
 Value EnvObject::refeq(Env *env, Value name, Value value) {
```

Comment:
`EnvObject::ref` and `EnvObject::refeq` have been fixed to handle corner cases per the spec tests.